threeML/plugins/FermiLATLLELike.py

```diff
@@ -0,0 +1,487 @@
+
+                self.triggertime = 0
+
+        self._bin_energies_into_pha()
+
+        with fits.open(ft2_file) as ft2_:
+
+            ft2_tstart = ft2_['SC_DATA'].data.field("START")  # - trigger_time
+            ft2_tstop = ft2_['SC_DATA'].data.field("STOP")  # - trigger_time
+            ft2_livetime = ft2_['SC_DATA'].data.field("LIVETIME")
+
+        ft2_bin_size = 1.0  # seconds
+
+        if not np.all(ft2_livetime <= 1.0):
+
+            warnings.warn("You are using a 30s FT2 file. You should use a 1s Ft2 file otherwise the livetime "
+                          "correction will not be accurate!")
+
+            ft2_bin_size = 30.0  # s
+
+        # Now we just need the live time fraction for the righ interval
+
+        livetime = ft2_livetime  # / (ft2_tstop - ft2_tstart)
+
+        # Keep only the needed entries (plus a padding)
+        idx = (ft2_tstart >= self._start_events - 10 * ft2_bin_size) & (
+            ft2_tstop <= self._stop_events + 10 * ft2_bin_size)
+
+        self._tstart = ft2_tstart[idx]
+        self._tstop = ft2_tstop[idx]
+        self._livetime = livetime[idx]
+
+        # Now sort all vectors
+        idx = np.argsort(self._tstart)
+
+        self._tstart = self._tstart[idx]
+        self._tstop = self._tstop[idx]
+        self._livetime = self._livetime[idx]
+
+    def _bin_energies_into_pha(self):
+        """
+
+        bins the LLE data into PHA channels
+
+        :return:
+        """
+
+        self._pha = np.zeros_like(self._energy, dtype=int)
+
+        for emin, emax, channel in zip(self._emin, self._emax, self._channels):
+
+            idx = np.logical_and(emin <= self._energy, self._energy < emax)
+
+            self._pha[idx] = channel
+
+        # There are some events outside of the energy bounds. We will dump those
+
+
+        self._filter_idx = self._pha > 0
+
+        self._n_channels = len(self._channels)
+
+    @property
+    def start_events(self):
+        return self._start_events
+
+    @property
+    def stop_events(self):
+        return self._stop_events
+
+    @property
+    def arrival_times(self):
+        return self._events[self._filter_idx]
+
+    @property
+    def energies(self):
+        return self._pha[self._filter_idx]
+
+    @property
+    def n_channels(self):
+        return self._n_channels
+
+    @property
+    def mission(self):
+        """
+        Return the name of the mission
+        :return:
+        """
+        return self._instrument
+
+    @property
+    def instrument(self):
+        """
+        Return the name of the instrument and detector
+
+        :return:
+        """
+
+        return self._telescope
+
+    @property
+    def livetime(self):
+        return self._livetime
+
+    @property
+    def livetime_start(self):
+        return self._tstart
+
+    @property
+    def livetime_stop(self):
+        return self._tstop
+
+    def _compute_mission_times(self):
```

Comment:
What is this for in this context? 

It is a nice functionality, maybe you could make it as a utility function under utils?